pkg/http/http.go

```diff
@@ -51,3 +52,29 @@ func GetURLResponse(url string, trim bool) (string, error) {
 
 	return respString, nil
 }
+
+func GetURLResponseWithTimeOut(url string, timeout time.Duration) ([]byte, error) {
+	client := http.Client{
+		Timeout: timeout,
+	}
+
+	// Will throw error as it's not quick enough
+	resp, httpErr := client.Get(url)
+	if httpErr != nil {
+		return nil, errors.Wrapf(httpErr, "an error occurred GET-ing %s", url)
+	}
+
+	defer resp.Body.Close()
+	statusOK := resp.StatusCode >= 200 && resp.StatusCode < 300
+	if !statusOK {
+		errMsg := fmt.Sprintf("HTTP status not OK (%v) for %s", resp.StatusCode, url)
+		return nil, errors.New(errMsg)
+	}
+
+	respBytes, ioErr := ioutil.ReadAll(resp.Body)
+	if ioErr != nil {
+		return nil, errors.Wrapf(ioErr, "could not handle the response body for %s", url)
+	}
+
+	return respBytes, nil
```

Comment:
Maybe a follow-up: refactor `GetURLResponse` to call `GetURLResponseWithTimeOut` but with a fixed value like `0`.